src/Compression/LZ4_decompress_faster.cpp

```diff
@@ -46,16 +46,58 @@ inline void copy8(UInt8 * dst, const UInt8 * src)
     memcpy(dst, src, 8);
 }
 
-inline void wildCopy8(UInt8 * dst, const UInt8 * src, const UInt8 * dst_end)
+
+inline void wildCopy8(UInt8 *dst, const UInt8 *src,
+                                 const UInt8 *dst_end)
 {
-    /// Unrolling with clang is doing >10% performance degrade.
-    #pragma nounroll
-    do
-    {
+    
+    /* -------- Small blocks: keep scalar -------- */
+    if (dst_end - dst < 128) {   
+        do {
+            copy8(dst, src);
+            dst += 8;  src += 8;
+        } while (dst < dst_end);
+        return;
+    }
+
+#if defined(__AVX512F__)        /* -------- AVX-512 path (64 B / iter) ------- */
+    const UInt8 *const orig_end = dst_end;
+    /* Align destination to 64 B for best store-forwarding behaviour        */
+    while (((uintptr_t)dst & 63) && dst < orig_end) {
+        copy8(dst, src);
+        dst += 8;  src += 8;
+    }
+
+    /* Main vectorised loop */
+    while (dst + 64 <= orig_end) {
```

Comment:
The name of the function, copy8, implies that it is called for small chunks of data.